orchestrator/graphql/schema.py

```diff
@@ -130,52 +135,64 @@ def custom_context_dependency(
 
 
 def get_context(
+    graphql_models: StrawberryModelType,
     broadcast_thread: ProcessDataBroadcastThread | None = None,
 ) -> Callable[[OrchestratorContext], Coroutine[Any, Any, OrchestratorContext]]:
     async def _get_context(
         custom_context: OrchestratorContext = Depends(custom_context_dependency),  # noqa: B008
     ) -> OrchestratorContext:
         custom_context.broadcast_thread = broadcast_thread
+        custom_context.graphql_models = graphql_models
         return custom_context
 
     return _get_context
 
 
-GRAPHQL_MODELS_INITIAL = dict(GRAPHQL_MODELS)
+INTERNAL_GRAPHQL_MODELS: StrawberryModelType = {}
+
+
+def internal_graphql_model_test_logic(graphql_models: dict) -> dict:
+    """Necessary Graphql logic for unit tests since we create and delete subscription models in each test and graphql expect static models.
+
+    We create and destroy the subscription models in each test. (which makes the pydantic model technically not the same after its re-created)
+    Strawberry will error saying that a nested product block can't be resolver and if you are actually using strawberry.field, This should never happen outside of testing.
+
+    returns INTERNAL_GRAPHQL_MOEDLS if environment is 'TESTING' and graphql_models if it isn't
+    """
+    if app_settings.ENVIRONMENT == "TESTING":
+        INTERNAL_GRAPHQL_MODELS.update(graphql_models)
+        return INTERNAL_GRAPHQL_MODELS
+    return graphql_models
 
 
 def create_graphql_router(
     query: Any = Query,
     mutation: Any = Mutation,
     register_models: bool = True,
-    subscription_interface: type | None = SubscriptionInterface,
+    subscription_interface: type = SubscriptionInterface,
     broadcast_thread: ProcessDataBroadcastThread | None = None,
+    graphql_models: StrawberryModelType | None = None,
+    scalar_overrides: ScalarOverrideType | None = None,
 ) -> OrchestratorGraphqlRouter:
-    @strawberry.experimental.pydantic.type(
-        model=SubscriptionModel, all_fields=True, directives=federation_key_directives
+    scalar_overrides = scalar_overrides if scalar_overrides else dict(SCALAR_OVERRIDES)
+    models = graphql_models if graphql_models else dict(DEFAULT_GRAPHQL_MODELS)
+    models["subscription"] = create_subscription_strawberry_type(
+        "Subscription", SubscriptionModel, subscription_interface
     )
-    class Subscription(SubscriptionInterface):
-        pass
-
-    models = dict(GRAPHQL_MODELS_INITIAL)
-    models["subscription"] = Subscription
 
     if register_models:
         models = register_domain_models(subscription_interface, existing_models=models)
-        GRAPHQL_MODELS.update(models)
+        models = internal_graphql_model_test_logic(models)
```

Comment:
Adding this fixture to `test/unit_tests/graphql/conftest.py` seem to remove the need for this. Does it work for you as well?

```python
from unittest import mock
from orchestrator.graphql.autoregistration import register_domain_models

@pytest.fixture(scope="session", autouse=True)
def fix_graphql_model_registration():
    internal_graphql_models = {}

    def patched_register_domain_models(*args, **kwargs):
        graphql_models = register_domain_models(*args, **kwargs)
        internal_graphql_models.update(graphql_models)
        return internal_graphql_models

    with mock.patch("orchestrator.graphql.schema.register_domain_models", side_effect=patched_register_domain_models):
        yield
```

I still don't understand why it's necessary, but better to have as little test-specific code in the actual code

Reply:
It has something to do with re-creating the classes per test, quick example:
```
CLASS_DICT = {}

def create_class():
    class Test:
        name: str = "1"

    CLASS_DICT['TEST'] = Test
    return Test()

test = create_class()
test2 = create_class()

print(isinstance(test, CLASS_DICT["TEST"]))  # this returns False
print(isinstance(test2, CLASS_DICT["TEST"]))  # this returns True
```